### PR TITLE
fix: added optional git reporting of pipelines for lighthouse

### PR DESCRIPTION
**Pull request by jstrachan**
this is a tactical solution; we are hoping longer term to use tekton PullRequestResource outputs instead

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -408,6 +411,10 @@ github.com/jbrukh/bayesian v0.0.0-20161210175230-bf3f261f9a9c h1:jqLMOnuwp+ac6nI
 github.com/jbrukh/bayesian v0.0.0-20161210175230-bf3f261f9a9c/go.mod h1:SELxwZQq/mPnfPCR2mchLmT4TQaPJvYtLcCtDWSM7vM=
 github.com/jenkins-x/draft-repo v0.0.0-20180417100212-2f66cc518135 h1:3zy/Nvdi9V95Jfu6+W4NAJrHDeypB58FSLyzI3XfO/4=
 github.com/jenkins-x/draft-repo v0.0.0-20180417100212-2f66cc518135/go.mod h1:K/L25ViEpDx196rOZyjn433tAM5zr2F/IouK+3g+DkE=
+github.com/jenkins-x/go-scm v1.5.20 h1:PDv9WNgZsG/sPwCdVkHJKBHE2jouLDFL2e5P5LyoQyU=
```
[the remark this answers is not in the document]
Comment (jstrachan) [reply go.sum]: I purposely went with go-scm to verify the code for the different git providers which we're gonna be using for lighthouse & to act as an example of what code we can use inside of tekton/lighthouse. We've also never really updated git statuses much inside jx itself yet (we assumed Jenkins / Prow would do that) so I've little clue how well tested any of that code is on non-github.

I can replace it with gitProvider code though if you'd rather though?

[the remark this answers is not in the document]
Comment (jstrachan) [reply go.sum]: BTW I'd also like to move `jx` to go-scm longer term as its a cleaner interface with more implementations and lots more tests - plus a nice fake implementation for testing

[the remark this answers is not in the document]
Comment (ccojocar) [reply go.sum]: I totally agree to moving towards `go-scm` in mid-term, but IMO is not a good example of maintainability if we try to mix things up. Unfortunately we did the same with `go-git`. It makes things more difficult: additional dependencies, different interfaces/abreactions, different testing approach, maintainability, confuses people which library to use etc. Personally, I believe that having some consistency across the board is helpful regardless of how cool a library it is.

[the remark this answers is not in the document]
Comment (pmuir) [reply go.sum]: In my opinion if we're going to introduce a competing depdency we then need to commit to changing everything to it. I agree with Cosmin that mixing dependencies up is a recipie for problems.

[the remark this answers is not in the document]
Comment (jstrachan) [reply go.sum]: OK have pushed a fix to reuse the gitProvider to update the status rather than go-scm

**Pull request merged**